src/kOS/Execution/Stack.cs

```diff
@@ -48,9 +48,13 @@ private void ThrowIfInvalid(object item)
         /// <returns></returns>
         private object ProcessItem(object item)
         {
-            if (item is float)
+            if (item is float) {
                 // promote floats to doubles
-                return Convert.ToDouble(item);
+                item = Convert.ToDouble (item);
+            }
+            if (item is double && (double)item == Math.Truncate((double)item)) {
+                return Convert.ToInt32 (item); // or should it just be (int)?
```

Comment:
suggest: 
```
if (item is double)
{
    double rounded = Math.Truncate((double)item);
    if ((double)item == rounded)
        return Convert.ToInt32(rounded);
}
```
Same thing, but it only performs the rounding once, instead of doing it in both Truncate and again in Convert.